core/metrics-core-service/src/test/java/org/hawkular/metrics/core/jobs/DeleteTenantITest.java

```diff
@@ -0,0 +1,302 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.core.jobs;
+
+import static java.util.Arrays.asList;
+
+import static org.hawkular.metrics.core.service.Order.ASC;
+import static org.hawkular.metrics.model.AvailabilityType.DOWN;
+import static org.hawkular.metrics.model.AvailabilityType.UP;
+import static org.hawkular.metrics.model.MetricType.AVAILABILITY;
+import static org.hawkular.metrics.model.MetricType.COUNTER;
+import static org.hawkular.metrics.model.MetricType.GAUGE;
+import static org.hawkular.metrics.model.MetricType.STRING;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.hawkular.metrics.core.service.BaseITest;
+import org.hawkular.metrics.core.service.DataAccess;
+import org.hawkular.metrics.core.service.DataAccessImpl;
+import org.hawkular.metrics.core.service.MetricsServiceImpl;
+import org.hawkular.metrics.model.AvailabilityType;
+import org.hawkular.metrics.model.DataPoint;
+import org.hawkular.metrics.model.Metric;
+import org.hawkular.metrics.model.MetricId;
+import org.hawkular.metrics.model.MetricType;
+import org.hawkular.metrics.model.Tenant;
+import org.hawkular.metrics.scheduler.api.JobDetails;
+import org.hawkular.metrics.scheduler.impl.TestScheduler;
+import org.hawkular.metrics.sysconfig.ConfigurationService;
+import org.jboss.logging.Logger;
+import org.joda.time.DateTime;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import com.codahale.metrics.MetricRegistry;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.google.common.collect.ImmutableMap;
+
+import rx.Observable;
+
+/**
+ * @author jsanda
+ */
+public class DeleteTenantITest extends BaseITest {
+
+    private static Logger logger = Logger.getLogger(DeleteTenantITest.class);
+
+    private MetricsServiceImpl metricsService;
+
+    private TestScheduler jobScheduler;
+
+    private JobsServiceImpl jobsService;
+
+    private static AtomicInteger tenantCounter;
+
+    private PreparedStatement getTags;
+
+    private PreparedStatement getRetentions;
+
+    private String jobName;
+
+    @BeforeClass
+    public void initClass() {
+        tenantCounter = new AtomicInteger();
+
+        getTags = session.prepare(
+                "SELECT tvalue, type, metric FROM metrics_tags_idx WHERE tenant_id = ? AND tname = ?");
+        getRetentions = session.prepare("SELECT metric FROM retentions_idx WHERE tenant_id = ? AND type = ?");
```

Comment:
Same here, (tenant_id, type) is enough for deletions and we have that information already. No need to fetch.